### PR TITLE
Add image for item inline with table of contents

### LINKED ISSUE #1599: Next to table of contents, add image for the item
The image size should probably scale in some useful way with the table of contents.

Here is a mockup for https://scholia.toolforge.org/topic/Q232169 .
Screen Shot 2021-08-12 at 19 18 30 copy

### PR DESCRIPTION
**Pull request by carlinmack**
Fix #1599

Add images to item pages inline with table of contents, making sure to resize appropriately. I used this comment to guide my approach, but didn't use any code https://stackoverflow.com/a/34402875

~~The `background-image` property is used as it's the only way to shrink an image to the size of it's neighbour without JS~~

image
image
image
    
### Caveats

> Please list anything which has been left out of this PR or which should be considered before this PR is accepted

~~For this to be accessible, the appropriate aria-roles should be placed so that it is clear that this is an image. Additionally, the `title=` attribute should be set so there is hover text and the image can be read out.~~ I briefly looked into getting the caption for the image from commons but it looked like I would need to make two API requests to get it. ~~When this `title` is added, there should be a fallback to the file name so that there is always a title set for the image.~~

Check any of the following which apply:

* [ ]  Breaking change (fix or feature that would cause existing functionality to not work as expected)
* [ ]  This change requires a documentation update
    * [ ]  I have made corresponding changes to the documentation
* [x]  This change requires new dependencies (please list)
   * I needed to introduce a way to get the MD5 of a string as the URL includes the first 2 characters from the MD5 of the file name. I used the following implementation from stackoverflow https://stackoverflow.com/a/60467595 They note it as free to use. 
   
### Testing
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

* Checked on several pages as shown above.
* Also checked an example item without an image

### Checklist
* [x]  I have commented my code, particularly in hard-to-understand areas
* [x]  My changes generate no new warnings

### CODE REVIEW

File: scholia/app/templates/base.html [hunk 1]
```diff
@@ -124,6 +124,23 @@
     });
     {% endif %}
 
+    if ("P18" in item.claims) {
+      console.log("in")
```
Comment (DanielFLopez) [inline scholia/app/templates/base.html:128]: I think you should remove the console.log

Comment (carlinmack) [reply scholia/app/templates/base.html:128]: yep sorry, this was actually why I added the ` * [ ] I have removed debug statements` to the PR template :)

**Comment by carlinmack** [comment]
Realised that as we are using JS anyway, we can set the height of the image correctly and we don't need to use the background image hack

**Review by egonw — CHANGES_REQUESTED** [review]
Looks nice! Can you check http://127.0.0.1:8100/author/Q18618629 ? I guess something doesn't go well with the character encoding?

**Comment by carlinmack** [comment]
It's actually an issue with the md5 function I chose. I've tried to comment on the answer to alert him of the problem but I need 50 upvotes on answers to be able to comment :/ I'll find a better implementation

**Comment by carlinmack** [comment]
Fixed 👍 there are not really any good options for unicode supporting md5 functions, without adding a crypto library

**Pull request merged**